### PR TITLE
Rafal dziuba/code source history

### PR DESCRIPTION
**Pull request by RafalDziuba**

### CODE REVIEW

File: src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue [hunk 1]
```diff
@@ -50,58 +83,29 @@ export default {
...
-        this.loaded = true;
-      });
-    } else {
-      // temporary until ArCode loads contracrt from the RedStone gateway
-      // if (this.isTestnet) {
-      axios.get(`${this.gatewayUrl}/gateway/contract-source?id=${this.sourceId}`).then((fetchedSource) => {
-        this.contractSrc = fetchedSource.data.src;
-        this.loaded = true;
+  async mounted() {
+    this.parseCode(this.source);
```
Comment (asiaziola) [inline src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue:99]: something weird is happening here, parseCode is an async function and you are not awaiting it here. additionally - you are parsing code even though another source code will be displayed. cant we call `await this.parseCode(this.preparedSource[0]);` after `preparedSource` is created?

File: src/pages/SmartWeaveExplorer/Contract/Contract.vue [hunk 2]
```diff
@@ -631,11 +647,12 @@ export default {
     },
 
     async getContract() {
-      const response = await fetch(`${this.gatewayUrl}/gateway/contract?txId=${this.contractId}`);
+      const response = await fetch(`https://gateway.warp.cc/gateway/v2/contract?txId=${this.contractId}`);
```
[the remark this answers is not in the document]
Comment (RafalDziuba) [reply src/pages/SmartWeaveExplorer/Contract/Contract.vue]: fixed

File: src/pages/SmartWeaveExplorer/Contract/Contract.vue [hunk 3]
```diff
@@ -631,11 +647,12 @@ export default {
...
       if (!response.ok) {
         this.correct = false;
       }
       const data = await response.json();
+      this.codeSource = data;
```
[the remark this answers is not in the document]
Comment (RafalDziuba) [reply src/pages/SmartWeaveExplorer/Contract/Contract.vue]: Fixed, but I thought that the transfer of the whole object does not affect performance

File: src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue [hunk 4]
```diff
@@ -50,58 +83,29 @@ export default {
...
+      activeItem: 0,
+      contractSrcHistory: [],
+      initSrcVersion: null,
+      preparedSource: [],
+      currentSrcTxId: null,
     };
   },
-  updated: function () {
-    Prism.highlightAll();
+  updated: async function () {
```
[the remark this answers is not in the document]
Comment (RafalDziuba) [reply src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue:94]: done

File: src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue [hunk 5]
```diff
@@ -1,45 +1,78 @@
...
-    <pre v-if="loaded && contractSrc && !wasm"><code class="language-javascript">{{contractSrc}}</code></pre>
-    <div v-if="loaded && contractSrc && wasm">
-      <ul id="code-wasm">
-        <li v-for="(item, idx) in contractSrc" :key="idx">
-          <pre class="py-4"><code>{{ idx.substring(idx.split('/', 4).join('/').length + 1) }}</code></pre>
-
-          <pre><code class="language-javascript">{{contractSrc[idx]}}</code></pre>
-        </li>
-      </ul>
+    <div class="version-nav" v-if="loaded && contractSrcHistory?.length > 0">
```
[the remark this answers is not in the document]
Comment (RafalDziuba) [reply src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue:3]: array is truthy even if it's empty

File: src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue [hunk 6]
```diff
@@ -50,58 +83,29 @@ export default {
...
-      //         const origin = event.origin;
-
-      //         const frameEvent = `${event.data.event}`.trim();
-      //         await this.handleCalls(frameEvent, event, contentWindow, origin);
-      //       },
-      //       false
-      //     );
-      //   }
     }
+    this.contractSrcHistory.push(this.source);
```
[the remark this answers is not in the document]
Comment (RafalDziuba) [reply src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue:105]: done

File: src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue [hunk 7]
```diff
@@ -239,16 +243,149 @@ export default {
...
+
+    async changeCodeSource(code, index) {
+      this.activeItem = index;
+      this.renderComponent = false;
+      await this.parseCode(code);
+      await this.$nextTick();
+      this.renderComponent = true;
+    },
+    convertTime: convertTime,
+    async prepareSource(source) {
```
[the remark this answers is not in the document]
Comment (RafalDziuba) [reply src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue:284]: done

File: src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue (hunk 1 above)
Comment (RafalDziuba) [reply src/pages/SmartWeaveExplorer/Contract/ContractCode/ContractCode.vue:99]: fixed

**Pull request merged**